src/Rings/msolve/msolve.jl

```diff
@@ -0,0 +1,209 @@
+ [52818775009509558395695966891//158456325028528675187087900672, 3//2535301200456458802993406410752, 845100400152152934331135470251//2535301200456458802993406410752]
+
+julia> msolve(I, get_param=true)
+((84*x^4 - 40*x^3 + x^2 + x, 336*x^3 - 120*x^2 + 2*x + 1, PolyElem[-184*x^3 + 80*x^2 - 4*x - 1, -36*x^3 + 18*x^2 - 2*x], BigInt[-1, -1]), Vector{fmpq}[[3197531698215911246794018079661//5070602400912917605986812821504, 1598765849107955623397009039829//5070602400912917605986812821504, -662230497759452443800611668909//5070602400912917605986812821504], [1, 0, 0], [143587366392252266220763399489//633825300114114700748351602688, 71793683196126133110381699745//633825300114114700748351602688, 173325283664805084153412401855//633825300114114700748351602688], [52818775009509558395695966891//158456325028528675187087900672, 3//2535301200456458802993406410752, 845100400152152934331135470251//2535301200456458802993406410752]])
+```
+"""
+function msolve(
+        I::MPolyIdeal;                        # input generators
+        initial_hts::Int=17,                  # hash table size, default 2^17
+        nr_thrds::Int=1,                      # number of threads
+        max_nr_pairs::Int=0,                  # number of pairs maximally chosen
+                                              # in symbolic preprocessing
+        la_option::Int=2,                     # linear algebra option
+        info_level::Int=0,                    # info level for print outs
+        precision::Int=32,                    # precision of the solution set
+        get_param::Bool=false                 # return rational parametrization of
+                                              # solution set
+        )
+    singular_assure(I)
+    SI    = I.gens.S
+    R     = I.gens.Sx
+    # skip zero generators in ideal
+    ptr = Singular.libSingular.id_Copy(SI.ptr, R.ptr)
+    J   = Singular.Ideal(R, ptr)
+    Singular.libSingular.idSkipZeroes(J.ptr)
+    # get number of variables
+    nr_vars = Singular.nvars(R)
+    nr_gens = Singular.ngens(J)
+    vars    = Singular.gens(R)
+
+    variable_names  = Array{String, 1}(undef, nr_vars)
+    for i in 1:nr_vars
+        variable_names[i] = string(Singular.gens(R)[i])
+    end
+
+    field_char  = Singular.characteristic(R)
+
+    #= add new variables and linear forms, =#
+    genericity_handling = 2
+
+    reset_ht  = 0
+    print_gb  = 0
+
+    #= monomial order defaults to zero =#
+    mon_order = 0
+
+    # convert Singular ideal to flattened arrays of ints
+    if 0 == field_char
+      lens, cfs, exps   = convert_qq_singular_ideal_to_array(J)
+    # elseif isprime(field_char)
+    #   lens, cfs, exps   = convert_ff_singular_ideal_to_array(J, nr_vars, nr_gens)
+    else
+        # error("At the moment GroebnerBasis only supports finite fields and the rationals.")
+        @error "At the moment msolve only supports the rationala as ground field."
```

Comment:
```suggestion
        @error "At the moment msolve only supports the rationals as ground field."
```
Why `@error`, though? This is for logging non-fatal errors, it seems to me you want to really throw an exception here (so e.g. using `error`)?